connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -363,7 +364,17 @@ public DistributedHerder(DistributedConfig config,
 
     @Override
     public void start() {
-        this.herderExecutor.submit(this);
+        Future<?> future = this.herderExecutor.submit(this);
+        try {
+            future.get(6, TimeUnit.SECONDS);
+        } catch (TimeoutException timeoutException) {
+            log.error("herder work thread timeout:", timeoutException);
+            future.cancel(true);
+        } catch (InterruptedException interruptedException) {
+            Thread.currentThread().interrupt();
+        } catch (ExecutionException executionException) {
+            log.error("herder work thread execution exception:", executionException);
```

Comment:
We don't want to change `start` to block. Can we save the future in a field in this class, then expose something like a `Future<?> herderTask()` method that allows testing code to access that future and, if desired, block on it?